manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/ServiceTypeHandler.java

```diff
@@ -48,6 +49,15 @@
             serviceTypeSnippet.setPath("spec.type");
 
             Boolean external = serviceSpec.get(HyscaleSpecFields.external, Boolean.class);
+            /**
+             * If loadBalancer is configured and external is specified as true in hspec, then Service type could be CLUSTER_IP.
+             * If loadBalancer is not configured, external is not specified or specified as false, then Service type could be CLUSTER_IP.
+             * Conclusion :  If loadBalancer is configured, service type will be CLUSTER_IP irrespective of the "external".
+             */
+            LoadBalancer loadBalancer = serviceSpec.get(HyscaleSpecFields.loadBalancer, LoadBalancer.class);
+            if (loadBalancer != null) {
+                external = false;
+            }
```

Comment:
We can combine the external and loadbalancer check in getServiceType(ServiceSpec) Method and move the comments to that method